woocommerce/woocommerce-android
# Issue/8389 subscriptions in product details screen

Partially closes: #8389

Partially closes: #8390

### Description


This PR displays the simple subscription details information on the product detail screen. It does that by extracting the subscription information from the product's metadata and adding a new subscription field to the product model. The new subscription field is used then to show the subscription information as one of the cards of the product's details.

### Testing instructions


1. Open the Products list screen
2. Select a simple subscription product
3. Check that the subscription card is shown among the product cards
4. Tap on the Subscription card
5. Check that the subscription details screen is shown

### Images/gif


https://user-images.githubusercontent.com/18119390/225412468-3fab8c40-66e4-475a-8f18-960de86ccf07.mp4
